### ISSUE TITLE
Conditional deploys ate Serverless-compose level

### ISSUE THREAD

**Issue by chris-hinds** [serverless/compose#104]

**Comment by tom-dq** [comment]
We would absolutely use a feature like this if it supported conditions based on region. At the moment we are doing this with environment variables, so as long as `--region` isn't supported by serverless compose it would be great if something like `if: ${env:region} == 'ap-southeast-2'` was also possible.

**Comment by mnapoli** [comment]
@tom-dq very interesting, thank you for providing another use case. That will be helpful if we start working on that to make sure the syntax/feature supports this.

To anyone reading this: I'd be interested to learn more about your use case. So far the typical use case we have is: "deploy service X only in production".

Let us know if you have other use cases.

**Comment by Hawki101** [comment]
We have a use case where we only need specific infrastructure deployed in one region. We currently do this by deploying our services separately with GH actions. It would be awesome to be able to define services that are only used in X regions. An example for us is our auth service which manages and contains our Cognito instance and some related hooks, which we currently share across regions.

**Comment by tom-dq** [comment]
Thanks @mnapoli - after reading @Hawki101's comment I should say, if serverless compose could natively support multi-region deploys, that would be a next-level developer experience for us. We have an architecture where most of the services are in multiple regions, but a few are global. So a great implementation for us would be:
 - Define a number of services, and the regions to which they should be deployed.
 - Allow the output of some services to be passed as a parameter to a dependent service, even if it's in another region.

I'm not sure if this has crept beyond the scope of the original issue - sorry if it has!

**Comment by Hawki101** [comment]
Yeah, this would be a better solution for us rather than deploying compose multiple times across regions and conditionally deploying services.

**Comment by stephenbawks** [comment]
I was also thinking about having a service/stack in my Compose file, that would allow to do deploy a feature branch for some infrastructure.  I obviously do not want the feature branch to run on normal PRs or merges, but thinking about running it when someone commits to a branch named `feature*`.  

I would also like to maybe append to the initial request because I could use this same `IF` statement so that I can conditionally deploy a stack as a feature branch.

**Comment by calebplum** [comment]
My use case is the opposite of what's described above - I have an authoriser Lambda function that protects non-prod environments. It's attached as a Lambda@Edge function so it must be deployed in us-east-1; since this is a different region to the rest of my application it must be deployed in a separate Serverless stack.

I'm using serverless-compose to orchestrate these two stacks, and I'm passing the ARN of the authoriser function to my main Serverless stack via serverless-compose parameters (to attach the authoriser to the Cloudfront distribution).

Since the production environment is public it doesn't require this authoriser function. It'd be great if serverless-compose supported conditional logic so that I could deploy the authoriser function only in my non-prod environments.

Currently my workaround involves using Cloudformation logic to prevent the authoriser **function** from being deployed in production, although serverless-compose still creates the Cloudformation stack for it, which feels messy. This also means I have to setup a dummy output in the authoriser's stack so that serverless-compose doesn't complain about missing parameters when I deploy to production, since the authoriser function's ARN is passed between stacks in my non-prod environments.

Conditional logic in serverless-compose would make this setup a whole lot cleaner and easier to manage.
